src/frontend/src/test-e2e/flows.ts

```diff
@@ -80,16 +80,11 @@ export const FLOWS = {
   },
   loginWelcomeView: async (
     userNumber: string,
-    deviceName: string,
     browser: WebdriverIO.Browser
   ): Promise<void> => {
     const welcomeView = new WelcomeView(browser);
     await welcomeView.waitForDisplay();
     await welcomeView.login(userNumber);
-    // This flow assumes no recovery phrase, so we explicitly skip the recovery nag here
-    await FLOWS.skipRecoveryNag(browser);
-    const mainView = new MainView(browser);
-    await mainView.waitForDeviceDisplay(deviceName);
```

Comment:
I added the await where `loginWelcomeView` is used.

I wanted to avoid coupling the logic of what comes after the flow from the flow.

Reply:
Another idea I had was to add a boolean parameter to control whether it should skip the recovery warning page or not.